### PR TITLE
Add unit tests for CLI and add flow compilation

### PR DESCRIPTION
**Pull request by ev1stensberg**
This PR enables us to test the webpack CLI as it stands in core, which in turns allows us to merge into the core repository

### CODE REVIEW

**Comment by ev1stensberg** [comment]
Once this gets a ok to merge, we'd need to remove the local import, but it's to emphasize that the CLI tests runs well locally if you clone and run the branch.

**Comment by ev1stensberg** [comment]
Could anyone review this?

File: package.json [hunk 1]
```diff
@@ -73,9 +77,13 @@
     "eslint-plugin-flowtype": "^2.35.1",
     "eslint-plugin-node": "^5.1.0",
     "flow-bin": "^0.49.1",
+    "flow-remove-types": "^1.2.1",
     "husky": "^0.14.3",
     "jest": "^20.0.4",
     "jest-cli": "^20.0.4",
-    "lint-staged": "^4.1.3"
+    "lint-staged": "^4.1.3",
+    "mocha": "^3.5.3",
```
Comment (okonet) [inline package.json]: We're using Jest for tests. Why is this needed?

File: package.json [hunk 2]
```diff
@@ -73,9 +77,13 @@
...
+    "prettier-eslint-cli": "^4.3.2",
+    "should": "^13.1.0"
```
Comment (okonet) [inline package.json]: Same, jest already has everything we need.

**Comment by ev1stensberg** [comment]
Yep, sure. Mocha was in old webpack, was easier to port those

**Comment by ev1stensberg** [comment]
@okonet done!

File: bin/convert-argv.js [hunk 3]
```diff
@@ -233,21 +242,22 @@ module.exports = function(yargs, argv, convertOptions) {
 		}
 
 		function mapArgToBoolean(name, optionName) {
+			if (options[optionName || name]) {
+				options[name] = true;
+			}
+			//eslint-disable-next-line
+			if (name && options[name] == true) {
```
Comment (ematipico) [inline bin/convert-argv.js]: Shouldn't be `===`?

Comment (ev1stensberg) [reply bin/convert-argv.js]: Implemented by webpack/core. You should just look at the test for now

Comment (ev1stensberg) [reply bin/convert-argv.js]: Also, let's keep it non-strict equal, cause it could be stringified

File: test/BinTestCases.test.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
+/* globals describe it before */
+"use strict";
+
+const path = require("path");
+const fs = require("fs");
+const spawn = require("child_process").spawn;
+
+function loadOptsFile(optsPath) {
+	// Options file parser from Mocha
```
Comment (ematipico) [inline test/BinTestCases.test.js]: Comment can be removed as we decided to stick with Jest

Comment (ev1stensberg) [reply test/BinTestCases.test.js]: Nice catch!

Comment (ev1stensberg) [reply test/BinTestCases.test.js]: Actually, this should be kept, if you see next line

File: bin/webpack.js [hunk 5]
```diff
@@ -1,36 +1,45 @@
...
 */
-"use strict";
-
 var path = require("path");
 var resolveCwd = require("resolve-cwd");
-
+// Local version replace global one
 var localCLI = resolveCwd.silent("webpack-cli/bin/webpack");
 
+if (process.argv.slice(2).includes("init")) {
```
Comment (ematipico) [inline bin/webpack.js]: `.includes` is not supported by Node 5 and 4. We need to change it. Also below
http://node.green/#ES2015-built-ins-well-known-symbols-Symbol-match--String-prototype-includes

**Comment by ematipico** [comment]
Some tests are failing on Node 5 and 4

**Comment by ev1stensberg** [comment]
Yep, cause we use `include`, changed in the `feature` PR

**Comment by ematipico** [comment]
Great, so it's under control :)

**Comment by ev1stensberg** [comment]
Any comments?

**Comment by ematipico** [comment]
It looks fine to me, there are just some conflicts but overall looks good

**Comment by ev1stensberg** [comment]
V4 failing cause there's too many async operations going on, LGTM for now I guess. @ematipico your opinion?

**Comment by ematipico** [comment]
Fine by me. We also added the removing of the types before publishing, which is awesome!
